slack_bot/app.py

```diff
@@ -77,18 +135,51 @@ def is_included_user(user: str, client: WebClient) -> bool:
         return True
 
     status_expiration = user_info["profile"]["status_expiration"]
-    tomorrow = datetime.datetime.now() + datetime.timedelta(days=1)
+    tomorrow = datetime.now() + timedelta(days=1)
     if status_expiration != 0 and status_expiration < tomorrow.timestamp():
         return True
 
     return False
 
 
-def get_chosen_users(all_users: list[str]) -> list:
-    return random.sample(all_users, k=2)
-
-
 def get_user_name(user_id: str, client: WebClient) -> str:
     user_info = client.users_info(user=user_id)
     user_name = user_info["user"]["profile"]["real_name"]
     return user_name.split()[0]
+
+
+def filter_users_based_on_previous_runs(users, previous_runs) -> list[str]:
+    if len(users) < 2:
+        return []
+
+    # Find the last run date of each user
+    last_run_dates = {user: max(run["date"] for run in previous_runs if user in run["pair"]) for user in users if
+                      any(user in run["pair"] for run in previous_runs)}
+
+    # Users who need a coffee break as they had none in the last 30 days
```

Comment:
da die files nach 30 tagen eh geloescht werden laut terraform-config, wie verhaelt sich das dann hier?

Reply:
Da die Dateien nach 30 Tagen gelöscht werden, wird hier nach Usern geschaut, die nicht in den Dateien vorhanden sind. Diese hatten dann in den letzten 30 Tagen auch keine Kaffeepause.